oxidation/libparsec_python/src/crypto.rs

```diff
@@ -16,8 +20,16 @@ impl HashDigest {
     }
 
     #[staticmethod]
-    fn from_data(data: &[u8]) -> PyResult<HashDigest> {
-        Ok(Self(parsec_api_crypto::HashDigest::from_data(data)))
+    fn from_data(py: Python, data: PyObject) -> PyResult<HashDigest> {
+        let copy;
+        let bytes = match data.extract::<&PyByteArray>(py) {
+            Ok(x) => {
+                copy = x.to_vec();
```

Comment:
Using [unsafe as_bytes](https://docs.rs/pyo3/0.15.1/pyo3/types/struct.PyByteArray.html#method.as_bytes) seems ok here given the GIL is held for the duration of the function and `&[u8]` doesn't leave the function body